crates/circuit/src/dag_circuit.rs

```diff
@@ -5843,12 +5786,16 @@ impl DAGCircuit {
                 )?
                 .into_any()
             }
-            NodeType::VarIn(var) => {
-                Py::new(py, DAGInNode::new(py, id, var.clone_ref(py)))?.into_any()
-            }
-            NodeType::VarOut(var) => {
-                Py::new(py, DAGOutNode::new(py, id, var.clone_ref(py)))?.into_any()
-            }
+            NodeType::VarIn(var) => Py::new(
+                py,
+                DAGInNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)),
+            )?
+            .into_any(),
+            NodeType::VarOut(var) => Py::new(
+                py,
+                DAGOutNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)),
+            )?
+            .into_any(),
```

Comment:
Is there any difference between using this and `into_py()`?

Reply:
I think `into_any` is the best choice here from an API perspective since it doesn't need a `Python` token. If we used `into_py` instead, then we'd be calling the impl of `IntoPy` for `Py<T>` which actually just calls `into_any` anyway ^^. So there's no difference, except I guess an extra function call which would almost certainly get optimized out anyhow.

https://docs.rs/pyo3/latest/src/pyo3/instance.rs.html#1719-1726

Reply:
Oh, my apologies for the confusion. What I meant was instead of declaring a `Py {*//*}`. We could just call `DAGOutNode.into_py(py)` I don't know if there are any limitations and why you're choosing your method. Just curious.